pkg/provider/aws/iam.go

```diff
@@ -25,17 +27,19 @@ func CreateIAMUserAndAttachPolicy(awsClient Client, username, policyArn *string)
 
 func CheckIAMUserExists(awsClient Client, username *string) (bool, error) {
 	if _, err := awsClient.GetUser(&iam.GetUserInput{UserName: username}); err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			// the specified IAM User doesn't exist
-			if aerr.Code() == iam.ErrCodeNoSuchEntityException {
-				return false, nil
-			}
+		var nse *types.NoSuchEntityException
+		if errors.As(err, &nse) {
+			return false, nil
 		}
 		return false, err
 	}
 	return true, nil
 }
 
+func Foo() int {
+	return 1
+}
+
```

Comment:
typo?

Reply:
😆 this was me testing something with my local config. good catch.